src/Jh/DataImportMagento/Writer/ShipmentWriter.php

```diff
@@ -45,25 +45,45 @@ public function writeItem(array $item)
         if (!isset($item['order_id'])) {
             throw new WriterException('order_id must be set');
         }
-
         $order = clone $this->orderModel;
         $order->loadByIncrementId($item['order_id']);
-
         if (!$order->getId()) {
             throw new WriterException(sprintf('Order with ID: "%s" cannot be found', $item['order_id']));
         }
 
+        $customerEmail = $order->getCustomerEmail();
+
         try {
             $shipment = $order->prepareShipment();
             $shipment->register();
             $shipment->getOrder()->setData('is_in_process', true);
 
+            if (isset($customerEmail)) {
```

Comment:
The e-mail sending should be behind an option like https://github.com/WeareJH/data-import-magento/blob/master/src/Jh/DataImportMagento/Writer/ReturnsWriter.php#L122. It should also be disabled by default otherwise it will break BC. Also use a null check not a isset check, the variable is always there, it just may be null.